arch/xtensa/include/xtensa-asm2-context.h

```diff
@@ -62,12 +64,25 @@
  * original/interrupted stack pointer.
  */
 
+#define BASE_SAVE_AREA_SIZE_COMMON	44
+
 #if XCHAL_HAVE_LOOPS
-#define BASE_SAVE_AREA_SIZE 56
+#define BASE_SAVE_AREA_SIZE_LOOPS	12
 #else
-#define BASE_SAVE_AREA_SIZE 44
+#define BASE_SAVE_AREA_SIZE_LOOPS	0
 #endif
 
+#if XCHAL_HAVE_S32C1I
+#define BASE_SAVE_AREA_SIZE_SCOMPARE	4
+#else
+#define BASE_SAVE_AREA_SIZE_SCOMPARE	0
+#endif
+
+#define BASE_SAVE_AREA_SIZE \
+	(BASE_SAVE_AREA_SIZE_COMMON + \
+	 BASE_SAVE_AREA_SIZE_LOOPS + \
+	 BASE_SAVE_AREA_SIZE_SCOMPARE)
```

Comment:
Yeah, this is a better way to compute that size than a giant #if tree of configuraiton variants.